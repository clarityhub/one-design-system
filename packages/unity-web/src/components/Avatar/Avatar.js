import React from 'react';
import styled from '@emotion/styled';
import { string } from 'prop-types';
import { Img } from 'react-image';

const AvatarWrapper = styled.span`
    display: inline-block;

    img {
        border-radius: 50%;
        height: 2.8rem;
        max-height: 2.8rem;
        max-width: 2.8rem;
        width: 2.8rem;
    }
`;

const Avatar = ({ alt = 'Avatar', url, fallbackUrl }) => (
	<AvatarWrapper>
		<Img
			src={[
				url,
				fallbackUrl,
			].filter(Boolean)}
			loader={fallbackUrl && <img alt={alt} src={fallbackUrl} />}
			alt={alt}
		/>
	</AvatarWrapper>
);

Avatar.propTypes = {
	alt: string,
	fallbackUrl: string.isRequired,
	url: string,
};

export default Avatar;
