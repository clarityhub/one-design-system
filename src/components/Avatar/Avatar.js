import React from 'react';
import styled from '@emotion/styled';
import { string } from 'prop-types';
import Img from 'react-image';

const AvatarWrapper = styled.span`
    display: inline-block;

    img {
        border-radius: 50%;
        height: 2.8rem;
        max-height: 2.8rem;
        max-width: 2.8rem;
        width: 2.8rem;
    }
`;

const Avatar = ({ alt = 'Avatar', url, fallbackUrl }) => (
    <AvatarWrapper>
        <Img
            src={[
                url,
                fallbackUrl,
            ].filter(Boolean)}
            loader={fallbackUrl && <img src={fallbackUrl} />}
            alt={alt}
        />
    </AvatarWrapper>
);

Avatar.propTypes = {
    alt: string,
    url: string,
    fallbackUrl: string.isRequired,
};

export default Avatar;
