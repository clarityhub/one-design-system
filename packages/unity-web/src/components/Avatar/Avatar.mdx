---
name: Avatar
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Avatar from './Avatar';

# Avatar

Use avatars to present user photographs.

You should always provide a `fallbackUrl` so that there is always
a preloaded image shown.

```js
import Avatar from '@clarityhub/unity-web/lib/components/Avatar';
```

## Avatars

Image with a fallback:

<Playground>
    <Avatar
        url="/public/ivan-avatar.png"
        fallbackUrl="/public/avatar-placeholder.png"
        alt="Ivan's Avatar"
    />
</Playground>

Image fails to load:

<Playground>
    <Avatar
        url="/public/invalid-url.png"
        fallbackUrl="/public/avatar-placeholder.png"
        alt="Ivan's Avatar"
    />
</Playground>


## Prop Types

The following props are available for the Avatar component.

<PropsTable of={Avatar} />