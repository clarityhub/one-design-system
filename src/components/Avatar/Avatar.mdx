---
name: Avatar
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Avatar from './Avatar';

# Avatar

## \<Avatar /\> component

```js
import Avatar from '@clarityhub/one/components/Avatar';
```

## Prop Types

<PropsTable of={Avatar} />


## Examples

<Playground>
    <Avatar
        url="/public/ivan-avatar.png"
        fallbackUrl="/public/avatar-placeholder.png"
        alt="Ivan's Avatar"
    />
</Playground>

<Playground>
    <Avatar
        url="/public/invalid-url.png"
        fallbackUrl="/public/avatar-placeholder.png"
        alt="Ivan's Avatar"
    />
</Playground>