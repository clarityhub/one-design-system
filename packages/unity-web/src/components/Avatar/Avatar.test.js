import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

describe('<Avatar />', () => {
	it('renders image', () => {
		const node = (
			<Avatar
				url="/public/ivan-avatar.png"
				fallbackUrl="/public/avatar-placeholder.png"
				alt="Ivan's Avatar"
			/>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
