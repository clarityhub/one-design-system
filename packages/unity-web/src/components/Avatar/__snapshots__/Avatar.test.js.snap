// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders image 1`] = `
<span
  className="css-anmw72-AvatarWrapper e1syn5zm0"
>
  <img
    alt="Ivan's Avatar"
    src="/public/avatar-placeholder.png"
  />
</span>
`;
